### ISSUE TITLE
Expectation for protocol in the binding uri

### ISSUE THREAD

**Issue by shawkins** [openshift/origin#17738]
The binding uri for pg and other databases includes the protocol:

https://github.com/openshift/origin/blob/de09446f1f3c8ff96da472e3ee71492a378db023/examples/db-templates/postgresql-ephemeral-template.json#L45

However this does not map directly to the needs of the various clients.  For JDBC you need postgesql://, for python/sqlalchemy you also need to specify the driver - postgresql+psycopg2://

Is the expectation that the bound service will need logic to modify the uri to turn it into the necessary url?  Or should the protocol be removed from the exposed uri?

**Comment by GrahamDumpleton** [comment]
My opinion so far is that the current variables injected by the service binding are actually quite useless. The chances of the names of the variables matching what frameworks/applications already expect in different languages is pretty slim. The current names and values, the form of which was possibly inherited from existing service broker world, may well work for a subset of Java applications but are totally useless for Python web applications where is more typical to accept a single ``DATABASE_URL`` environment variable which includes a full URI, where login/password and database name are also in the URI.

So personally I can't see how can recommend using service bindings at all, and making suggestions for changes is probably fruitless as is probably working to some existing convention that cannot be changed, and whatever you change it to, even if doesn't break existing usage, will never satisfy a majority of users anyway because there is going to be too many different conventions. So better to continue to manually set up environment variables yourself based on your specific application requirements.

**Comment by shawkins** [comment]
I agree that the current binding have a lot of limitations.  The only thought is that it would better to just have the uri containing host:port without the protocol.  That also may not match many existing applications, but it would be broadly applicable via simple substitution with most clients.

**Comment by GrahamDumpleton** [comment]
The convention that existed in OpenShift 2 was better. Separate fields for all entities. With them separate, you can at least compose them together in different ways easily, and don't have to rip apart composite fields to get out separate values for host and port as have to now if need to pass them separately. You could still then have separate variable which has a full URI with all fields as can be fit in the URI and not leave bits out such as login, password and database.

**Comment by shawkins** [comment]
Bringing back the separate fields would also be fine.

**Comment by rareddy** [comment]
@bparees This issue is along with #17230 getting to be cumbersome to provide further solutions to enhance the consumption of database properties in platfoms like RHOAR. The issues are
* URI is not usable (in the case of Postgres it is wrong) in any of sources it is defined. It can be kept as is provided below is corrected.
* Service name can not be found from bound properties in credentials, thus user is forced to parse the URI which is not standard nor usable directly. User is forced to parse this URI to find service name, even then you get the IP of the Pod rather than service name.

I will submit a PR for these corrections for your review.

**Comment by rareddy** [comment]
captured at software collections  https://github.com/sclorg/postgresql-container/issues/275
